sklearn/cluster/_birch.py

```diff
@@ -324,13 +324,26 @@ def radius(self):
 
 
 class Birch(ClusterMixin, TransformerMixin, BaseEstimator):
-    """Implements the Birch clustering algorithm.
+    """Implements the BIRCH clustering algorithm.
 
     It is a memory-efficient, online-learning algorithm provided as an
     alternative to :class:`MiniBatchKMeans`. It constructs a tree
     data structure with the cluster centroids being read off the leaf.
     These can be either the final cluster centroids or can be provided as input
     to another clustering algorithm such as :class:`AgglomerativeClustering`.
+    
+    This is not a complete implementation of the original BIRCH algorithm.
+    It differs from BIRCH in the following aspects:
+    
+    - there is no memory limit implemented, and the threshold is never adapted
+      to prevent the tree from becoming too large.
+      Instead, the user has to choose a desirable threshold in advance.
+    - the diameter criterion is not supported, only radius is supported
+    - only Euclidean distance (D0) is supported, D1-D4 are not supported
+    - handling of outliers is not implemented
+    - Phase 2 (tree condensation) is not implemented
+    - Phase 3 clustering uses only the centroids of the leaves
```

Comment:
Might be worth noting: as opposed to what?

I recall that I'd always wanted to weight the reclustering. I think a major reason that we didn't at the time is that scikit-learn lacked clustering implementations that supported sample_weight.

Reply:
The weight and the variance information contained in the leaves.
Why bother storing the variance, weight etc. when its then not further used later on?

Reply:
I'm not sure how a new user would understand these points. Users are not necessarily very familiar with the original paper of our methods. I think this is a fair point to mention. I'm thinking whether it'd require explaining what BIRCH is supposed to do and then explain the differences, maybe in the user guide?

Reply:
A lot of people will only look at the class documentation, not the user guide.  So at least a warning should be placed in the class documentation that this implementation differs quite a lot from BIRCH.

Reply:
We could have a warning and say, this implementation differs in certain ways from the original paper, please look at <ref> for more info?

Reply:
We don't want docstrings to be too detailed, so +1 for directing readers to the UG for more info about the implem (in general we always have "Read more in the User Guide" at the end of each docstring).

Reply:
I don't think a runtime warning is appropriate. I agree that documentation is better.